src/MarkSheet.java

```diff
@@ -0,0 +1,261 @@
+			computeGrades();
+			printStatistics();
+		}
+		else
+		{
+			System.out.println("Invalid grade values are present in the passed array");
+		}
+	}
+		
+	MarkSheet()
+	{
+		/*
+		 * Default unparameterized constructor
+		 * Used when grades value are not passed
+		 */
+		grades=new float[getCount()];
+		fetchDetails();
+		computeGrades();
+		printStatistics();
+	}
+	
+	public boolean valudateinput(float[] value)
+	{
+		/*
+		 * validate the grades value and return true if array is valid.
+		 * valid grades value are in range 0 to 100 both inclusive.
+		 */
+		for(int a=0;a<value.length;a++)
+		{
+			if(a>100||a<0)
+			{
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	private static int getCount()
+	{
+		/*
+		 * Used for Taking the student count as input from the user
+		 * Takes count of students as input
+		 * validation of input is performed for Mismatch data type
+		 */
+		int count=0;
+		System.out.println("Enter no of students : ");
+		try
+		{
+			count=scanner.nextInt();
+			return count;
+		}
+		catch(InputMismatchException e)
+		{
+			System.out.println("Please enter Integral value");
+			scanner.nextLine();
+			count=getCount();
+		}
+		return count;
+	}
+
+	private static void fetchDetails() 
+	{
+		/*
+		 * Takes input from helper function and store the value of grade received in grades array.
+		 * Validation of input is performed in helper funtion
+		 */
+		
+		for(int temp=0;temp<grades.length;temp++)
+		{
+			grades[temp]=getGrade(temp);
+		}
+	}
+
+	private static float getGrade(int n)
+	{
+		/*
+		 * Helper function for fetchDetails
+		 * Taking the input from the user
+		 * Takes count of students as parameters
+		 * validation of input is performed as grade can have values between 0 to 100 both inclusive.
+		 * checking of input data type is also done.
+		 */
+		float grade=0f;
+		try
+		{
+			System.out.print("Enter the grade for student "+(n+1)+" : ");
+			grade=scanner.nextFloat();
+			if(grade>=0&&grade<=100)
+			{
+				return grade;
+			}
+			else
+			{
+				System.out.println("Enter vaues in range 0 to 100");
+				grade=getGrade(n);
+			}
+		}
+		catch(InputMismatchException e)
+		{
+			System.out.println("Enter numerical value only ");
+			scanner.nextLine();
+			grade=getGrade(n);
+		}
+		return grade;
+	}
+
+	private static void computeGrades()
+	{
+		/*
+		 * Traverse throughout the grades array and calculate.
+		 * calculate the number of students who achieve above or atleast 40% grade.
+		 * initially minimum grade is set to be the maximum value a grade can have.
+		 * initially maximum grade is set to be the minimum value a grade can have.
+		 * Sum of grade is also calculated for calculation of Average later on.
+		 * Working is used so the function have to fetch the value of a grade multiple times.
+		 * 
+		 */
+		int pass_count=0;
+		float max_grade=0;
+		float min_grade=100;
+		float grade_sum=0;
+		float working;
+		for(int temp=0;temp<grades.length;temp++)
+		{
+			working=grades[temp];
```

Comment:
Instead of pre validate the input you could do the validation here also. By doing so you could save one iteration of the whole array.